### PR TITLE
i18n: developer documentation

### PR DESCRIPTION
**Pull request by dachary**
## Status

Ready for review

## Description of Changes

Developer oriented documentation for i18n.

## Deployment

Only documentation, no deployment side effect.

### If you made changes to documentation:

- [x] Doc linting passed locally

### CODE REVIEW

File: docs/development/i18n.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+Internationalization AKA i18n
+=============================
+
+The code, templates and javascript user visible strings must all be wrapped with a `gettext`_ functions to be substituted at runtime with the equivalent localized string. These function names are used as markers to collect the strings to be translated and store them into the ``securedrop/translations/messages.pot`` file. For each language to be translated, a directory is created such as ``securedrop/translations/fr_FR`` and populated with files derived from ``securedrop/translations/messages.pot``, for translators to work with and for the `gettext`_ substitution at runtime.
```
Comment (redshiftzero) [inline docs/development/i18n.rst]: "with a" -> "with"

Comment (redshiftzero) [inline docs/development/i18n.rst]: In line with the [documentation guidelines](https://docs.securedrop.org/en/latest/development/documentation_guidelines.html) can we wrap these lines in this PR to ~80 characters? I know that's kind of annoying but it does make line-by-line review clearer

File: docs/development/i18n.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+The `pybabel`_ command line is wrapped into the ``manage.py translate`` helper for convenience. It is designed to be used by developers, to run tests with fixtures and during packaging.
+
+Updating strings to be translated
+---------------------------------
+
+After modifying a string in the code, templates or javascript, the ``securedrop/translations/messages.pot`` file must be updated by running the following command at the root of the git tree, in the development virtual machine::
+
+    ./manage.py --verbose translate --extract-update
+
+The updated ``securedrop/translations/messages.pot`` should then be reviewed and commited. Once merged, the changes will be visible in the `weblate`_ web interface used by translators.
```
Comment (redshiftzero) [inline docs/development/i18n.rst]: `weblate` -> `Weblate`

File: docs/development/i18n.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+Compiling translations
+----------------------
+
+The `gettext`_ need a compiled file for each language (the ``*.mo`` files). This can be done by running the following command at the root of the git tree, in the development virtual machine::
```
Comment (redshiftzero) [inline docs/development/i18n.rst]: The `gettext`_ need -> `gettext`_ needs

File: docs/development/i18n.rst (hunk 2 above)
Comment (redshiftzero) [inline docs/development/i18n.rst]: Does that updating happen automatically (is Weblate configured to watch `develop` in this repo?) or does it require some kind of further interaction?

File: docs/development/i18n.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+After modifying a string in the code, templates or javascript, the ``securedrop/translations/messages.pot`` file must be updated by running the following command at the root of the git tree, in the development virtual machine::
+
+    ./manage.py --verbose translate --extract-update
+
+The updated ``securedrop/translations/messages.pot`` should then be reviewed and commited. Once merged, the changes will be visible in the `weblate`_ web interface used by translators.
+
+Compiling translations
+----------------------
+
```
Comment (redshiftzero) [inline docs/development/i18n.rst]: Can we add some comments to describe once translations for a target language are completed how they are exported from Weblate?

File: docs/development/i18n.rst [hunk 5]
```diff
@@ -0,0 +1,48 @@
+Internationalization AKA i18n
```
Comment (redshiftzero) [inline docs/development/i18n.rst]: "Internationalization AKA i18n" -> "Internationalization (i18n)"

File: docs/development/i18n.rst [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+Verifying translations
+----------------------
+
+After a translation is compiled, the web page in which it shows can be verified visually by navigating to the corresponding state from ``http://localhost:8080`` for the source interface or ``http://localhost:8081`` for the journalist interface after running the following::
+
+     ./manage.py run
+
+An easier way is to generate screenshots for each state in each known language with::
+
+     pytest -v tests/pages-layout
```
Comment (redshiftzero) [inline docs/development/i18n.rst]: In this case for clarity one can use `$ pytest -v tests/pages-layout` to clearly denote that this is a command and the text that follows is output of that command

File: docs/development/i18n.rst [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+     ...
+     ...TestJournalistLayout::test_col_no_documents[en_US] PASSED
+     ...TestJournalistLayout::test_col_no_documents[fr_FR] PASSED
+     ...
+
+The screenshots for ``fr_FR`` are available in ``securedrop/tests/pages-layout/screenshots/fr_FR`` and the name of the file can be found in the function that created it in ``securedrop/tests/pages-layout/test_journalist.py`` or ``securedropt/tests/pages-layout/test_source.py``.
```
Comment (redshiftzero) [inline docs/development/i18n.rst]: `securedropt/tests/pages-layout/test_source.py` -> `securedrop/tests/pages-layout/test_source.py`

File: docs/development/i18n.rst (hunk 5 above)
Comment (dachary) [inline docs/development/i18n.rst]: done

File: docs/development/i18n.rst (hunk 1 above)
Comment (dachary) [inline docs/development/i18n.rst]: ah right ! done

File: docs/development/i18n.rst (hunk 2 above)
Comment (dachary) [inline docs/development/i18n.rst]: I'll add to the documentation to answer that question.

File: docs/development/i18n.rst (hunk 4 above)
Comment (dachary) [inline docs/development/i18n.rst]: yes, good point, added a **Merging translations** section

File: docs/development/i18n.rst (hunk 3 above)
Comment (dachary) [inline docs/development/i18n.rst]: done

File: docs/development/i18n.rst (hunk 6 above)
Comment (dachary) [inline docs/development/i18n.rst]: ah shoot, forgot about that, thanks for the reminder

File: docs/development/i18n.rst (hunk 7 above)
Comment (dachary) [inline docs/development/i18n.rst]: good catch, done

**Comment by dachary** [comment]
rebased & updated & repushed. Thanks for the careful review !

**Comment by dachary** [comment]
rebased to resolve conflict & repushed & make docs-lint again

**Comment by dachary** [comment]
rebased & repushed. Thanks to my newly acquired privileges, ci/circleci will run. I hope ?

**Comment by dachary** [comment]
continued at https://github.com/freedomofpress/securedrop/pull/2118

**Pull request closed without merging**